src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp

```diff
@@ -443,14 +443,9 @@ void SyncInferRequest::wait() {
         } else if (is_remote_tensor_impl && is_dynamic) {
             auto& stream = m_graph->get_network()->get_stream();
             auto user_mem = remote_tensor_impl_ptr->get_original_memory();
-            if (user_mem->get_allocation_type() == cldnn::allocation_type::cl_mem && output_memory->get_allocation_type() != cldnn::allocation_type::cl_mem) {
-                // WA: Copy between cl_mem and usm memory may fail for some reason (driver bug?)
-                // so this explicit memcpy is used to provide correct output for cl_mem output in dynamic cases
-                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::write> lock_dst(user_mem, stream);
-                cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::read> lock_src(output_memory, stream);
-                std::memcpy(lock_dst.data(), lock_src.data(), output_memory->size());
-            } else {
-                copy_events.push_back(output_memory->copy_to(stream, *user_mem, false));
+            if (!m_graph->get_engine().is_the_same_buffer(*output_memory, *user_mem)) {
+                auto blocking = (stream.get_queue_type() == QueueTypes::in_order) ? false : true;
+                copy_events.push_back(output_memory->copy_to(stream, *user_mem, blocking));
```

Comment:
Could you please clarify why out of order queue needs to use blocking copy here?

Reply:
blocking has been removed